.github/workflows/docker-publish.yml

```diff
@@ -84,7 +87,7 @@ jobs:
       # transparency data even for private images, pass --force to cosign below.
       # https://github.com/sigstore/cosign
       - name: Sign the published Docker image
-        if: ${{ github.event_name != 'pull_request' }}
+        if: ${{ github.event_name != 'pull_request' && contains(github.event.head_commit, 'Dockerfile')}}
```

Comment:
Uma vez que o objetivo do processo de gerar a imagem é somente se o Dockerifle for alterado, o IF pode ficar no começo das steps, ai nenhuma seria executada e teríamos somente um IF.